btrbk: 0.22.2 -> 0.24.0

kampka opened the pull request:
###### Motivation for this change

Upgrades btrbk to the latest upstream version.
This new version requires config migrations due to [incompatible changes](https://github.com/digint/btrbk/blob/master/doc/upgrade_to_v0.23.0.md).
Does a package need to handle this somehow?

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

kampka committed:
btrbk: 0.22.2 -> 0.24.0

the-kenny commented:
Thanks for the heads-up! I've got some personal backup services that depend on it, but I'm fine with modifying them. Grepping for `btrbk` doesn't yield any services depending on it.

kampka commented:
@the-kenny right, I also didn't find any service, so I created [my own](https://gist.github.com/kampka/fbe40826d0a9bf38884d9bad192f56ff).
It is kind of opinionated at the moment, but I could be persuaded to improve on it an send a pull request.

kampka commented:
I added openssh to the path in order to get the ssh integration working.
Since the change seems trivial, I squashed the commit ontop of the version bump.

the-kenny commented:
Merged as a1d9d7efc0
